pkg/services/oauthserver/oauthimpl/service.go

```diff
@@ -297,22 +299,55 @@ func (s *OAuth2ServiceImpl) handleKeyOptions(ctx context.Context, keyOption *oau
 		return nil, nil
 	}
 
+	keyType := "ecdsa"
```

Comment:
it's a bit strange to force it in here and yet have a switch case on it after

Reply:
I'm working on this and I have the change on my local, but I couldn't finish it because of the security issue I had to handle. :)